mediasession.bs

```
@@ -263,8 +269,53 @@ session state">idle</a></code>.
 The <dfn attribute for="MediaSession"><code>kind</code></dfn>
 attribute must return the <a>media session</a>'s <a>kind</a>.
 
+The <dfn method for="MediaSession"><code>activate()</code></dfn> method, when
+invoked, must run these steps:
+
+<ol>
+  <li>
+    Let <var>media session</var> be the <a>context object</a>.
+  </li>
+  <li>
+    Let <var>promise</var> be a new promise.
+  </li>
+  <li>
+    Return <var>promise</var>, and run the remaining steps <a>in parallel</a>.
+  </li>
+  <li>
+    <a>Activate</a> <var>media session</var>.
+  </li>
+  <li>
+    If <a>activate</a> failed, reject <var>promise</var> with a
+    <code>TypeError</code>.
```

I think we will only have one mode of failure for `activate()` as Android doesn't give us any extra information when `AudioManager.requestAudioFocus()` fails. That being said, I think that using `DOMException` of a somewhat appropriate type is entirely harmless given that many existing APIs does so.

@domenic, I think you're more involved with ECMA stuff, is there a precedent to follow when designing new APIs, and is `TypeError` the catch-all exception to be using?